test/integration/transaction.test.js

```diff
@@ -176,6 +176,146 @@ describe(Support.getTestDialectTeaser('Transaction'), function() {
         });
       });
 
+      it('supports for update of table', function() {
+        var User = this.sequelize.define('User', { username: Support.Sequelize.STRING })
+          , Task = this.sequelize.define('Task', { title: Support.Sequelize.STRING, active: Support.Sequelize.BOOLEAN })
+          , self = this
+          , test = false;
+
+        User.belongsToMany(Task, { through: 'UserTasks' });
+        Task.belongsToMany(User, { through: 'UserTasks' });
+
+        return this.sequelize.sync({ force: true }).then(function() {
+          return Promise.join(
+            User.create({ username: 'John'}),
+            Task.create({ title: 'Get rich', active: false}),
+            Task.create({ title: 'Die trying', active: false}),
+          function (john, task1, task2) {
+            return john.setTasks([task1]);
+          }).then(function() {
+            return self.sequelize.transaction(function(t1) {
+              return User.find({
+                where: {
+                  username: 'John'
+                },
+                include: [Task]
+              }, {
+                lock: t1.LOCK.UPDATE,
+                transaction: t1
+              }).catch(function(err) {
+                expect(err.message).to.be.equal('FOR UPDATE cannot be applied to the nullable side of an outer join');
```

Comment:
You might want to split this  test into two - one that check the error, and one that checks that update is not blocked.

That'd also remove the ugly ` test=true`, because you can simply:

```js
return expect(User.find ...).to.be.rejectedWith('FOR UPDATE cannot be applied to the nullable side of an
```